Update Kontrol from `0.1.316` to `1.0.53`

JuanCoRo opened the pull request:
<!--
Contributions welcome! See https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md
-->

**Description**
This PR updates Kontrol to version `1.0.53` making the following changes:
- Updates `make-summary-deployment.sh` and `run-kontrol.sh` scripts for the newer version
- Removes the `kdeploy` profile since now generate the summaries directly from the `Deploy.s.sol`
- Updates the `copy_memory_to_memory` summary lemma
- Points all proof files except `OptimismPortal.k.sol` to use the `DeploymentSummaryFaultProofs` instead of `DeploymentSummary`
- Removes the `kontrol/deployment` folder

JuanCoRo committed:
versions.json: update Kontrol from `0.1.316` to `1.0.37`

JuanCoRo committed:
Remove `kdeploy` profile

JuanCoRo committed:
Update kontrol commands

JuanCoRo committed:
Run Fault Proofs summary by default

JuanCoRo committed:
Update non fault proofs summaries

JuanCoRo committed:
Update fault proofs summaries

JuanCoRo committed:
Remove `deployment` folder

JuanCoRo committed:
Exclude `proveWithdrawalTransaction` proofs

JuanCoRo committed:
Revert "Exclude `proveWithdrawalTransaction` proofs"

This reverts commit 5dd0f0393ff654fa46c7ca7d5cd02e03ffd12b79.

JuanCoRo committed:
`run-kontrol.sh`: add `--bmc-depth 1`

JuanCoRo committed:
foundry.toml: remove `kdeploy` profile

JuanCoRo committed:
.gitignore: remove `kout-deployment`

JuanCoRo committed:
Merge remote-tracking branch 'origin/develop' into update-kontrol-v1-0-37

JuanCoRo committed:
Update summaries

JuanCoRo committed:
`common.sh`: account for `FOUNDRY_PROFILE` being `default`

mds1 commented on packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh:
```diff
@@ -36,9 +36,10 @@ kontrol_prove() {
     --kore-rpc-command 'kore-rpc-booster --equation-max-recursion 100' \
-    --xml-test-report
+    --xml-test-report \
+    --bmc-depth 1
```
> Introduces `--bmc-depth 1` as part of the `kontrol prove` options as a workaround until the `copy_memory_to_memory` lemma is updated

What's the summary on why this resolves the failures, and are there risks/tradeoffs here?

JuanCoRo replied:
It's a way of not fully executing the loops (in this case, the one in `copy_memory_to_memory`). Instead of fully summarizing the loop's behavior, we only execute one iteration. The tradeoff is that it's a less rigorous reasoning technique than having the summary in place.
Raoul's [excellent recent post](https://runtimeverification.com/blog/formally-verifying-loops-part-1) explains this in fantastic detail!

JuanCoRo replied:
Resolving this since the `copy_memory_to_memory` summary has been updated!

JuanCoRo committed:
Remove `check-kontrol-deployment.sh`

Since now the Kontrol snapshot is taken from the `Deploy.s.sol` script, there is
no need to check that the custom Kontrol deployment is unmodified. Plus, the
custom Kontrol deployment process was removed previously on this PR

JuanCoRo committed:
justfile: remove `kontrol-deployment-check` from running checks

JuanCoRo committed:
config.yml: remove `kontrol-deployment-check`

Commit:
moving to Kontrol v1.0.50, invariant operational

Commit:
always rekompile and regen

Commit:
workers correction

Commit:
moving to Kontrol v1.0.51

JuanCoRo committed:
run-kontrol.sh: shellcheck typos

JuanCoRo committed:
run-kontrol.sh: add `verbose` and flags for `rekompile` and `regen`

JuanCoRo committed:
Merge remote-tracking branch 'origin/develop' into update-kontrol-v1-0-37

JuanCoRo committed:
run-kontrol.sh: remove `--verbose` for `kontrol prove`

JuanCoRo committed:
Update `DeploymentSummaryFault` code

JuanCoRo committed:
Update `DeploymentSummary`

JuanCoRo committed:
Merge remote-tracking branch 'origin/develop' into update-kontrol-v1-0-37

JuanCoRo committed:
Update Kontrol to version `1.0.53`

smartcontracts commented on packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh:
```diff
@@ -36,9 +37,15 @@ kontrol_prove() {
-    --init-node-from $state_diff \
-    --kore-rpc-command 'kore-rpc-booster --equation-max-recursion 100' \
-    --xml-test-report
+    --init-node-from-diff $state_diff \
```
Are there docs that explain these options?

JuanCoRo replied:
There are [these docs](https://github.com/runtimeverification/kontrol/tree/master/docs/external-computation), explaining the previous version. Initially, the only option was `from-diff`, but we now support the `dumpState` cheatcode, so we need to differentiate between the `diff` and the `dump`. Documentation on how to get the state updates from the `dumpState` cheatcode can be found [in this tutorial](https://github.com/runtimeverification/kontrol-workshop-ethcc-7/tree/main/4-external-computation), although I haven't managed to update everything together yet

smartcontracts commented on packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh:
```diff
@@ -107,10 +114,9 @@ lemmas=test/kontrol/pausability-lemmas.md
 module=OptimismPortalKontrol:$base_module
 rekompile=--rekompile
-rekompile=
+# rekompile=
```
Is there a reason to include this line at all?

JuanCoRo replied:
Just for convenience. As described above, unsetting the variable will remove that option, so the comment is there to remove two characters instead of writing the whole thing. I'm not opposed to remove this if you don't like it!
Also note that we now support toml specifications for this options, so in the future all of this might be just contained in the `kontrol.toml` file

smartcontracts replied:
Cool, seems fine to me!

smartcontracts commented on packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh:
```diff
@@ -107,10 +114,9 @@ lemmas=test/kontrol/pausability-lemmas.md
 regen=--regen
-# shellcheck disable=SC2034
-regen=
+# regen=
```
Same question as above

JuanCoRo replied:
Same answer as above :+1:

smartcontracts commented on packages/contracts-bedrock/test/kontrol/pausability-lemmas.md:
```diff
@@ -202,17 +202,17 @@ The summary lemma is as follows, with commentary inlined:
-       andThenBool SRC +Int LENGTH <=Int DEST // No overlap between source and destination
-       andThenBool DEST <=Int lengthBytes(LM) // Destination starts within current memory
-       andThenBool PCOUNT +Int 51 <Int lengthBytes(JUMPDESTS) // We're not look outside of the JUMPDESTs bytearray
+       andBool 0 <=Int PCOUNT
```
What’s the difference between `andThenBool` vs `andBool`?

JuanCoRo replied:
This was introduced to silence an error that appeared when Kontrol tried to access an array outside of its bounds. This error was caused by the `requires` clauses not being applied in order since the bound constraint was introduced before the array access. `andThenBool` was put in place to force the ordering of the constraints. However, this should have been ordered automatically. The error appears to be showing no longer.
Also, note that the error of out-of-bounds array access did not affect the soundness of the proof. It just made the summary not apply, which shouldn't have applied anyway in those cases.

smartcontracts reviewed (commented):
Couple of small questions/remarks
